operator/buffer/memory.go

```diff
@@ -29,36 +31,135 @@ func NewMemoryBufferConfig() *MemoryBufferConfig {
 	}
 }
 
-// Build builds a MemoryBufferConfig into a Buffer, loading any entries that were previously unflushed
-// back into memory
-func (c MemoryBufferConfig) Build(context operator.BuildContext, pluginID string) (Buffer, error) {
-	return &MemoryBuffer{}, nil
+// Build builds a MemoryBufferConfig into a Buffer
+func (c MemoryBufferConfig) Build(_ operator.BuildContext, pluginID string) (Buffer, error) {
+	return &MemoryBuffer{
+		pluginID:      pluginID,
```

Comment:
Is this supposed to assign a buffer to each operator? If that's the intent of the ID, we should change this to operatorID, since plugins are a different concept in stanza. I know it's confusing, because operators seem like essentially plugins.

Reply:
Yeah that makes sense. I was think too the build context doesn't need to be passed in.

Reply:
It also doesn't need to return an error; Is this implementing some interface? It seems like it's trying vaguely to be similar to the `Builder` interface, but it doesn't fit that either.

Reply:
Scratch that; It IS implementing a `Builder` interface, the one that's in this package.

Reply:
Yeah, I think it use to return an error when it loaded from disk. The disk buffer likely will have a need for this interface returning an error.